pkg/ddl/ingest/checkpoint.go

```diff
@@ -389,24 +416,24 @@ func (s *CheckpointManager) updateCheckpointLoop() {
 	defer ticker.Stop()
 	for {
 		select {
-		case wg := <-s.updaterCh:
+		case finishCh := <-s.updaterCh:
 			err := s.updateCheckpoint()
 			if err != nil {
-				logutil.DDLIngestLogger().Error("update checkpoint failed", zap.Error(err))
+				s.logger.Error("update checkpoint failed", zap.Error(err))
 			}
-			wg.Done()
+			close(finishCh)
 		case <-ticker.C:
 			s.mu.Lock()
-			if !s.dirty || s.updating {
```

Comment:
If this is removed, is it possible to update checkpoint to a smaller value? For example,

1. g1 invokes updateCheckpoint, current checkpoint is 1.
2. g2 invokes updateCheckpoint, current checkpoint is 2.
3. g2 updates checkpoint 2 to table.
4. g1 updates checkpoint 1 to table.

Reply:
I see `updateCheckpoint` is only called by one goroutine which is `updateCheckpointLoop`